src/platform/backends/lxd/lxd_virtual_machine.cpp

```diff
@@ -344,9 +353,14 @@ const QUrl mp::LXDVirtualMachine::network_leases_url()
     return base_url.toString() + "/networks/" + bridge_name + "/leases";
 }
 
-void mp::LXDVirtualMachine::request_state(const QString& new_state)
+void mp::LXDVirtualMachine::request_state(const QString& new_state, const QJsonObject& state_options)
 {
-    const QJsonObject state_json{{"action", new_state}};
+    QJsonObject state_json{{"action", new_state}};
+
+    for (auto option = state_options.constBegin(); option != state_options.constEnd(); ++option)
+    {
+        state_json.insert(option.key(), option.value());
+    }
```

Comment:
QJsonObject feels like it's playing against you here, wouldn't a simpler container be easier to deal with?

Reply:
Well, yeah, except I'm having to send a `QJsonObject` to LXD here and found that just using a `QJsonObject` from the beginning was the simplest way to go.  Otherwise, we'd still need to convert from simpler container to `QJsonObject` here via some iterator.  The sad thing is that `QJsonObject` just doesn't have a good way to concatenate 2 `QJsonObject`s :frowning:

Reply:
Just after posting the above, I thought of a way to simplify this some.  I can do something like:
```
QJsonObject state_json{state_options};

state_json.insert("action", new_state);
```

That should avoid the iterator and still accomplish the same thing.

Reply:
Ok, I changed to the way I proposed above.